packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -402,6 +412,11 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
                     }
                 };
 
+                connection.onNotification('onClose', event => {
+                    this.isIdle = true;
+                    this.term.writeln('\x1b[1m\n\rTerminal will be reused by tasks. \x1b[0m\n');
```

Comment:
Terminals should know nothing about tasks. It breaks abstractions, such code should belong to tasks extensions.

You don't need `onClose` notification, there is already `connection.onDispose` and `connection.onClose` on the json-rpc level.